ext/sqlite3/sqlite3.c

```diff
@@ -1218,10 +1241,15 @@ PHP_METHOD(sqlite3, openBlob)
 
 	sqlite3_stream = emalloc(sizeof(php_stream_sqlite3_data));
 	sqlite3_stream->blob = blob;
+	sqlite3_stream->flags = flags;
 	sqlite3_stream->position = 0;
 	sqlite3_stream->size = sqlite3_blob_bytes(blob);
 
-	stream = php_stream_alloc(&php_stream_sqlite3_ops, sqlite3_stream, 0, "rb");
+	if (flags != 0) {
+		mode = "r+b";
+	}
```

Comment:
OK changed that to use the constants SQLITE3_OPEN_*, it looks better like that you're right :) Also added tests for errors.